### ISSUE TITLE
Move all logging/debug messaging function to one location

### ISSUE THREAD

**Issue by hansenms** [gadgetron/gadgetron#14]
We should have a uniform set of debug functions. The functions introduced with gtPlus are in the wrong location and header files need to be merged.

**Comment by hansenms** [comment]
First state of this has been done. Remaining stuff:

* Getting the timing statements to come out properly in the log.
* Replace all std::cout statements with proper log statement.
* Add some comments to the file and add a wiki page with guidelines on how to log stuff in the gadgetron. 
* Adding log configuration to the gadgetron.xml configuration file to allow config for a given installation. 
* Adding some more levels (like a VERBOSE and some custom channels for debugging specific gadgets)
* Option to send log to file.
* Option to return log to client if used in the Gadgetron. This is going to require a bit more work, because we need to separate logs from potentially multiple Gadget chains running in parallel. I need to do some thinking for this.
* Logging thread indices. Some thought needs to be put into this to make it cross platform compatible. 
Potentially adding a "module" label to the output. Basically that would be the name of the shared library emitting the log statement.

**Comment by hansenms** [comment]
I am closing this issue for now, since the initial goal has been met. I will open a new issue with enhancements for the logging system that we may want to see.
